skytemple_randomizer/randomizer/util/util.py

```diff
@@ -499,23 +507,25 @@ def get_main_string_file(
     # If we didn't find english, just take the first
     if lang is None:
         lang = static_data.string_index_data.languages[0]
-    return lang, FileType.STR.deserialize(
-        rom.getFileByName(f"MESSAGE/{lang.filename}"),
-        string_encoding=static_data.string_encoding,
-    )
+    return lang, get_lang_string_file(rom, static_data, lang)
 
 
 def get_all_string_files(
     rom: NintendoDSRom, static_data: Pmd2Data
 ) -> Iterable[tuple[Pmd2Language, Str]]:
     for lang in static_data.string_index_data.languages:
-        yield (
-            lang,
-            FileType.STR.deserialize(
-                rom.getFileByName(f"MESSAGE/{lang.filename}"),
-                string_encoding=static_data.string_encoding,
-            ),
+        yield lang, get_lang_string_file(rom, static_data, lang)
+
+
+def get_lang_string_file(
```

Comment:
The two get methods for `Str` have been modified to use this common `get_lang_string_file` instead - this allows for this third method to be used for `get_pokemon_name` below.